supervisor/shared/web_workflow/web_workflow.c

```diff
@@ -529,7 +552,13 @@ static void _reply_redirect(socketpool_socket_obj_t *socket, _request *request,
         _send_str(socket, "http");
     }
 
-    _send_strs(socket, "://", hostname, ".local", path, "\r\n", NULL);
+    _send_strs(socket, "://", hostname, ".local", NULL);
+    if (web_api_port != 80) {
+        char encoded_port[6];
+        snprintf(encoded_port, sizeof(encoded_port), "%d", web_api_port);
+        _send_strs(socket, ":", encoded_port, NULL);
+    }
```

Comment:
It occurs to me that it would be possible to implement `_send_print` so that you could write `mp_plat_print(&_send_print, ":%d", web_api_port)` here.

Reply:
Do you know if mp_plat_print will allocate to the MP heap? I think that's why I avoided it.